analysis/webservice/algorithms/doms/ResultsStorage.py

```diff
@@ -361,7 +377,7 @@ def __retrieveStats(self, id):
             }
             return stats
 
-        raise Exception("Execution not found with id '%s'" % id)
+        raise NexusProcessingException(reason=f'Execution not found with id {str(execution_id)}', code=404)
```

Comment:
Is this wording accurate? The execution may be found but the data may not be present